### ISSUE TITLE
`.clone().takes_self_ref()` isn't caught by clippy::redundant_clone

### ISSUE THREAD

**Issue by kangalio** [rust-lang/rust-clippy#9455]
### Summary

Code of the form `.clone().method()`, where `method` takes `&self`, has a redundant clone. Clippy should catch this

### Lint Name

redundant_clone

### Reproducer

I tried this code:

```rust
#[derive(Clone)]
struct Foo;
impl Foo {
    fn method(&self) {}
}

fn example(foo: &Foo) {
    foo.clone().method();
}
```

I expected to see this happen: a warning that the `.clone()` call is redundant

Instead, this happened: no such warning

### Version

```text
rustc 1.63.0 (4b91a6ea7 2022-08-08)
binary: rustc
commit-hash: 4b91a6ea7258a947e59c6522cd5898e7c0a6a88f
commit-date: 2022-08-08
host: x86_64-unknown-linux-gnu
release: 1.63.0
LLVM version: 14.0.5
```
